test/react/providers/room-provider.test.tsx

```diff
@@ -0,0 +1,162 @@
+        </ChatClientProvider>
+      );
+    };
+    render(<TestProvider />);
+
+    // Try to get the client to get a room with different options, should fail
+    expect(() => chatClient.rooms.get(roomId, RoomOptionsDefaults)).toThrowErrorInfoWithCode(40000);
+
+    // Now try it with the right options, should be fine
+    expect(() => chatClient.rooms.get(roomId, { reactions: RoomOptionsDefaults.reactions })).toBeTruthy();
+  });
+
+  it('it should correctly release rooms', () => {
+    const chatClient = newChatClient() as unknown as ChatClient;
+    const TestComponent = () => {
+      return <div />;
+    };
+    const roomId = randomRoomId();
+    const TestProvider = () => {
+      return (
+        <ChatClientProvider client={chatClient}>
+          <RoomProvider
+            id={roomId}
+            options={{ reactions: RoomOptionsDefaults.reactions }}
+            attach={false}
+            release={true}
+          >
+            <TestComponent />
+          </RoomProvider>
+        </ChatClientProvider>
+      );
+    };
+    const r = render(<TestProvider />);
+
+    // Try to get the client to get a room with different options, should fail
+    expect(() => chatClient.rooms.get(roomId, RoomOptionsDefaults)).toThrowErrorInfoWithCode(40000);
+
+    // Now try it with the right options, should be fine
+    expect(() => chatClient.rooms.get(roomId, { reactions: RoomOptionsDefaults.reactions }));
+
+    // Unmount provider
+    r.unmount();
+
+    // Since the room is supposed to be released on unmount, we should be able
+    // to get it again with different settings
+    expect(() => chatClient.rooms.get(roomId, RoomOptionsDefaults)).toBeTruthy();
+  });
+
+  it('it should attach and detach correctly', () => {
+    const chatClient = newChatClient() as unknown as ChatClient;
+    const TestComponent = () => {
+      return <div />;
+    };
+    const roomId = randomRoomId();
+
+    const room = chatClient.rooms.get(roomId, { reactions: RoomOptionsDefaults.reactions });
+    expect(room).toBeTruthy();
+
+    vi.spyOn(room, 'attach');
+    vi.spyOn(room, 'detach');
+
+    const TestProvider = () => {
+      return (
+        <ChatClientProvider client={chatClient}>
+          <RoomProvider
+            id={roomId}
+            options={{ reactions: RoomOptionsDefaults.reactions }}
+            attach={true}
+            release={false}
+          >
+            <TestComponent />
+          </RoomProvider>
+        </ChatClientProvider>
+      );
+    };
+    const r = render(<TestProvider />);
+
+    // Make sure the room is attaching
+    expect(room.attach).toHaveBeenCalled();
+
+    r.unmount();
+    // Make sure the room is detaching
+    expect(room.detach).toHaveBeenCalled();
+
+    // Try to get the client to get a room with different options, should fail
+    expect(() => chatClient.rooms.get(roomId, RoomOptionsDefaults)).toThrowErrorInfoWithCode(40000);
+
+    void chatClient.rooms.release(roomId);
```

Comment:
What are we testing here, is this necessary? :) If we want to make sure we release the room, why are we voiding here?

Reply:
Actually not sure it's necessary, not testing anything with this line.

Reply:
removed